helix-core/src/textobject.rs

```diff
@@ -74,13 +74,7 @@ pub fn textobject_word(
     match textobject {
         TextObject::Inside => Range::new(word_start, word_end),
         TextObject::Around => Range::new(
-            match slice
-                .get_char(word_start.saturating_sub(1))
-                .map(categorize_char)
-            {
-                None | Some(CharCategory::Eol) => word_start,
-                _ => prev_grapheme_boundary(slice, word_start),
-            },
```

Comment:
This breaks selecting the whitespace to the left if there is none to select to the right, for example a word at the end of a line:

```rust
// cursor anywhere on "here" and then `maw`

// expected:
some text[ here]

// current:
some text [here]
```

The reasoning is that "around" a word should delete the whole unit of the word which includes the whitespace.

Reply:
Ah, I misunderstood again, it seems.  So, to confirm, what you want is:

1. If there is a space on the right, include that in the selection.
2. Only if that's *not* the case, then select a space to the left.

?

Reply:
Also: it's only spaces, right?  So it should never select additional surrounding characters other than spaces?

Reply:
Okay, so I've changed it so it will only select around if there's whitespace available, and will select to the left only if there's no whitespace to the right.

Let me know if that's the desired behavior.

Reply:
Ah I forgot to mention, it should select _all_ of the whitespace to the left or right; using the previous example,

```rust
// cursor on here
some text[    here]

// cursor on text
some [text    ]here
```

> Also: it's only spaces, right? So it should never select additional surrounding characters other than spaces?

Yeah, only spaces.

Reply:
Ah!  Got it.  I'll change that next.  Thanks for the help!

Reply:
Okay, it should have the desired behavior now.

Reply:
It works correctly now :)